### DISCUSSION TITLE
Devlink based firmware update mechanism for wwan modules

### DISCUSSION THREAD

**Discussion by jsomashe** [fwupd/fwupd#4918] [Q&A]
Hello Team,

Is develink based firmware update mechanism supported by fwupd ?

Best Regards, Somashekar

**Comment by hughsie** [comment]
I suspect this might answer the question, but what's _develink_?

**Reply by jsomashe** [reply]
The devlink-flash API allows updating device firmware. It replaces the older ethtool-flash mechanism, and doesn’t require taking any networking locks in the kernel to perform the flash update. Example use:
$ devlink dev flash pci/0000:05:00.0 file flash-boot.bin
https://www.kernel.org/doc/html/latest/networking/devlink/devlink-flash.html
Similar to fastboot way of updating wwan modules, need help to understand if devlink based flashing plugin support is possible in fwupd ?

**Reply by hughsie** [reply]
Ahh, thanks for the link. It's not supported by fwupd right now, but it looks like it would be fairly easy to add support for. I think somebody (not me!) would need to write a plugin which implements the `info`, `flash` and `reboot` client side API in fwupd.

**Reply by jsomashe** [reply]
Thank you for confirming.
